### PR TITLE
add list of regexes useful for redacting from build logs

### PR DESCRIPTION
**Pull request by PeterDSteinberg**
Addresses #157, giving the ability to redact the progress bars from the build log.  

If ```quiet: True``` is given in the .binstar.yml for a package submission, the read_with_timeout function will now check each line against a list of regexes called QUIET_REGEXES to see if the line should be redacted from the build log.  Currently the only regex in QUIET_REGEXES is a one that matches patterns like:
```
ncurses-5.9-1.  22% |#######                        | ETA:  0:00:00 100.51 MB/s
```

### CODE REVIEW

File: binstar_build_client/worker/utils/timeout.py [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
 import logging
+import re
+import time
 
 log = logging.getLogger('binstar.build')
 
 
+QUIET_REGEXES = [
+    re.compile('(\|\s*#[#.\s]*\|\s*ETA)'),
+]
```
Comment (PeterDSteinberg) [inline binstar_build_client/worker/utils/timeout.py]: If we have more ideas of patterns to remove from the build log, we can put them in this list.

**Comment by PeterDSteinberg** [comment]
The second commit in this PR addresses a string vs bytes issue in Python 3 regexes.  The third commit fixes the new test on windows Python 3. 

The Python 2 Windows test is failing, but appears to be unrelated to the PR.

**Comment by srossross** [comment]
@PeterDSteinberg do you think a better solution would be to collapse the lines edning with `\r` when displayed on the server?

**Comment by srossross** [comment]
Here: https://github.com/Anaconda-Server/anaconda-server/blob/develop/binstar/sites/www/views/profile/builds.py#L212

**Comment by PeterDSteinberg** [comment]
Sounds like a good idea @srossross .  I had forgotten about \r - that does seem more efficient and general.

**Comment by PeterDSteinberg** [comment]
This PR is just failing on Windows Python 2.7 for reasons unrelated to this PR.

**Comment by stephenakearns** [comment]
Hi @PeterDSteinberg  - is there an issue somewhere that tracks the problem that your'e blocked on? If yes, can you add the link? If not, can you make sure it's logged so that we can address it?

**Comment by PeterDSteinberg** [comment]
@stephenakearns The issue of failing build worker seen here in Windows Python 2.7 relates to anaconda-build issue #150 . I don't think it should hold up merging this PR.

**Comment by dsludwig** [comment]
It works for me, I just wonder if redacting from the log is the right solution. It might be nicer if it shows up collapsed in the UI. For example, implemented this way someone running `build tail -f` will not see progress bars show up until they are complete.

**Comment by PeterDSteinberg** [comment]
My vote is that we merge this PR for the ```quiet: True``` option for now, and then later when we do the work to develop better annotated build logs, as described in issue #86, at that time we can come up with a saved separate build log item for the \r messages (that issue will use something like echo tags in build scripts to better delineate in the UI which part of build log comes from which statement).  I think the progress bars are equally annoying in tail -f as in viewing logs on the server UI.

**Comment by srossross** [comment]
> My vote is that we merge this PR for the quiet: True option for now

:+1:

**Pull request merged**